### ISSUE TITLE
i met Nan problem after  training 1 epoch.

### ISSUE THREAD

**Issue by zfyong** [vita-epfl/openpifpaf#16]
i flow the official steps to train openpifpaf. at first epoch the training loss is ok, but it appear Nan at second training epoch. here is training logs:
{"message": null, "levelname": "INFO", "name": "Trainer", "asctime": "2019-03-21 15:24:33,960", "type": "train", "epoch": 1, "batch": 1620, "n_batches": 7074, "time": 0.564, "data_time": 0.0, "lr": 0.001, "loss": NaN, "head_losses": [NaN, NaN, 0.387, NaN, NaN, NaN]}

**Comment by svenkreiss** [comment]
The Google Drive also includes the logs of our training runs. At the top of the log file, all command line parameters are listed. Can you compare with yours?

Our log files are also visualized (training and validation losses etc) in the included notebook: https://github.com/vita-epfl/openpifpaf/blob/master/docs/studies.ipynb

**Comment by svenkreiss** [comment]
Was just comparing notes myself and found three missing command line flags in the readme command for training. Please add:

```sh
  --momentum=0.95 \
  --quad=1 \
  --crop-fraction=0.5 \
```

I am updating the readme now.

**Comment by svenkreiss** [comment]
Thanks for your report and for pointing out this problem. I hope this is fixed now for you.
I am closing this issue. If this did not fix your problem, please re-open.

**Comment by svenkreiss** [comment]
I am seeing this now as well for ResNet50 models. Investigating the burn-in/warm-up rate. Originally it was 0.1 but the latest default is 0.01 which might be too low for ResNet50 but required for ResNet152 models. Investigating.

**Comment by CHUNYUWANG** [comment]
I still see this problem even after about 20 epochs of training. Have you found the reason for NaN?

**Comment by svenkreiss** [comment]
I have not seen this problem anymore after the above fix (#21). Can you share your training command?

**Comment by CHUNYUWANG** [comment]
I tried to train a ResNet-18 model (which means I modified the code to add support for Resnet-18). I am not sure if I myself introduced the problem or not. 

In my case, the third component of the loss is also NOT NaN just as @zfyong 

I noticed that you removed the code of clamping the predicted spread. Is it possible this is related to NaN?

The command is as follows:

python3 -m openpifpaf.train \
  --lr=1e-3 \
  --momentum=0.95 \
  --epochs=75 \
  --lr-decay 60 70 \
  --batch-size=8 \
  --basenet=resnet18 \
  --head-quad=1 \
  --headnets pif paf \
  --square-edge=401 \
  --regression-loss=laplace \
  --lambdas 30 2 2 50 3 3 \
  --freeze-base=0

**Comment by svenkreiss** [comment]
With different backbones, different training parameters might be necessary. When you encounter nan's it means that your learning is too aggressive. Try to reduce the learning rate and increase the batch size.

Check for example a log file for the pre-trained shufflenet models for ideas. The first line of the log file contains the training parameters. It's very different from the ResNet-50/101/152 parameters. And shufflenetv2x2 does not use ImageNet pretraining which also influences the training.

**Comment by CHUNYUWANG** [comment]
Thanks for your detailed feedback! I currently tend to think this error is related to my CUDA environment. After upgrading cuda from 9 to 10, the problem of NaN disappears although I don't know why.
